docs/getting-started/tutorials/table_api.md

```diff
@@ -0,0 +1,250 @@
+Both tables support running batch and streaming applications.
+
+#### Registering A UDF
+
+Along with tables we include a user defined function for working with timestamps. Our function takes a timestamp and rounds it down the nearest hour. 
+
+{% highlight java %}
+tEnv.registerFunction("truncateDateToHour", new TruncateDateToHour());
+{% endhighlight %}
+
+#### The Query
+
+With our environment configured and tables registered we are ready to build our first application.
+From the `TableEnvironment` we can `scan` an input table to read its rows and then write those results into an output table using `insertInto`. 
+
+{% highlight java %}
+tEnv
+    .scan("transactions")
+    .insertInto("stdout");
+{% endhighlight %}
+
+This intial job simply reads all transactions and writes them to standard out. 
+
+#### Execute
+
+Flink applications are built lazily shipped to the cluster for execution only once fully formed. 
+We call `ExecutionEnvironment#execute` to begin the execution of our job. 
+
+{% highlight java %}
+env.execute("Spend Report");
+{% endhighlight %}
+
+## Attempt One
+
+Now that we have the skeleton of a job, let's add some business logic.
+We want a report that shows the total spend for each account across each hour of the day.
+Just like a SQL query, we can select the required fields and group by our keys. 
+Because the timestamp field has millisecond granularity we will use our UDF to round it down to the nearest hour.
+Finally, we will select all the fields, summing the total spend per account-hour pair.
+
+{% highlight java %}
+tEnv
+    .scan("transactions")
+    .select("accountId, timestamp.truncateDateToHour as timestamp, amount")
+    .groupBy("accountId, timestamp")
+    .select("accountId, timestamp, amount.sum as total")
+    .insertInto("stdout");
+{% endhighlight %}
+
+## Adding Windows
+
+While this works, we can do better.
+The `timestamp` column represents the [event time]({{ site.baseurl }}/dev/event_time.html) of each row, where event time is the logical time when an event took place in the real world.
+Flink understands the concept of event time, which we can leverage to clean up our code. 
+
+Bucketing data based on time is very common operation in data processing, especially when working with infinite streams. 
+A bucket based on time is called a `Window` and Flink offers flexible windowing semantics. 
+The most basic type of window is called a `Tumble` window, which has a fixed size and whose buckets do not overlap. 
+
+{% highlight java %}
+tEnv
+    .scan("transactions")
+    .window(Tumble.over("1.hour").on("timestamp").as("w"))
+    .groupBy("accountId, w")
+    .select("accountId, w.start as timestamp, amount.sum")
+    .insertInto("stdout");
+{% endhighlight %}
+
+This defines our application as using one hour tumbling windows based on the timestamp column.
+So a row with timestamp `2019-06-01 01:23:47` will be put in the `2019-06-01 01:00:00` window.
+While semantically equivalent, when using windows:
```

Comment:
This list of reasons for preferring windows is the section I am most unsure about. I'm am writing to an audience that has no idea what a window is and I'm not sure this clearly expresses the benefits.

Reply:
Something like: Aggregations based on time are special, because time (as opposed to other arbitrary attributes) generally moves forward in a continuous streaming application. This enables Flink to perform certain optimizations such as state clean up, when using time window-based aggregations.